metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/MerkleTree.scala

```diff
@@ -117,6 +117,9 @@ object MerkleTree {
       }
     }
 
+    val maxLeafIndex = (1 << proof.siblingPath.length) - 1
```

Comment:
This is required so for example on a single element tree a `Proof(4, Nil)` is not accepted, only `Proof(0, Nil)`, otherwise it would mean the `CheckpointCertificate` is malleable by adding arbitrarily large numbers as long as the LSB side that the verification uses is correct.

Reply:
That looks like excellent comment to be placed in code.

Reply:
Done.